### PR TITLE
ZkPrograms QoL improvements

### PR DESCRIPTION
**Pull request by mitschabaude**
Small collection of QoL improvements around ZkProgram:
* change the `TupleToInstances<>` type so that it gets fully resolved by intellisense, i.e. we now see a normal method signature when hovering over ZkProgram methods
* Expose `maxProofsVerified()` and the `Proof` class directly on the ZkProgram
* Fix an outdated ZkProgram test
* Delete an unused test/example (that was also outdated and didn't build)

### CODE REVIEW

**Comment by mitschabaude** [comment]
Ready for review

**Comment by mitschabaude** [comment]
note: the commonjs test failure just seems to be a runner problem

**Review by Geometer1729 — APPROVED** [review]
LGTM

**Comment by Geometer1729** [comment]
Your first two bullets should probably be added to CHANGELOG.md

CI is supposed to catch when PRs don't update the changelog, not sure why it didn't here.

File: src/examples/zkprogram/mututal-recursion.ts [hunk 1]
```diff
@@ -47,15 +47,13 @@ const add = ZkProgram({
...
-const AddProof = ZkProgram.Proof(add);
-
 const multiply = ZkProgram({
   name: 'multiply',
   publicInput: Undefined,
   publicOutput: Field,
   methods: {
     performMultiplication: {
-      privateInputs: [Field, AddProof],
+      privateInputs: [Field, add.Proof],
```
[the remark this answers is not in the document]
Comment (Trivo25) [reply src/examples/zkprogram/mututal-recursion.ts:56]: TS/VSCode sometimes takes a while to apply the freshly compiled changes - simply delete the import statement, save the file and re-import it. At least this fixes it for me

[the remark this answers is not in the document]
Comment (45930) [reply src/examples/zkprogram/mututal-recursion.ts:56]: Restarting my TS server didn't help, but quitting and re-openeing VS Code fixed, so disregard.

File: src/lib/proof-system/zkprogram.ts [hunk 2]
```diff
@@ -1112,7 +1126,7 @@ type Infer<T> = T extends Subclass<typeof ProofBase>
 
 type TupleToInstances<T> = {
   [I in keyof T]: Infer<T[I]>;
-} & any[];
+};
```
Comment (45930) [inline src/lib/proof-system/zkprogram.ts:1129]: This is really nice! <3

**Review by 45930 — APPROVED** [review]

**Review by Trivo25 — APPROVED** [review]
That's a great and clean addition! I think this is good to go if we can add a quick changelog entry too!

**Comment by mitschabaude** [comment]
updated the PR incl. changelog, just waiting for base PR to merge

**Pull request merged**